### PR TITLE
Fixes to the bug of categories.order never being initialized

### PR DESCRIPTION
**Pull request by kotodharma**

### CODE REVIEW

File: app/controllers/categories_controller.rb [hunk 1]
```diff
@@ -26,7 +26,7 @@ def edit
   def create
     @category = Category.new(category_params)
 
-    if @category.save
```
Comment (jonpage) [inline app/controllers/categories_controller.rb]: using before_save would be a little more railsy: http://api.rubyonrails.org/classes/ActiveRecord/Callbacks.html

File: app/models/category.rb [hunk 2]
```diff
@@ -3,6 +3,16 @@ class Category < ActiveRecord::Base
 
   belongs_to :data_list
 
+  def store_cat
+    self.order = 0
+    # Even if parent_id is nil this does the right thing.
+    last_child = Category.where(ancestry: parent_id).order('categories.order desc').limit(1)[0]
+    if last_child && last_child.order
```
Comment (jonpage) [inline app/models/category.rb]: `list_order` is the relevant column for this story

File: app/models/category.rb [hunk 3]
```diff
@@ -3,6 +3,16 @@ class Category < ActiveRecord::Base
 
   belongs_to :data_list
 
+  def store_cat
+    self.order = 0
+    # Even if parent_id is nil this does the right thing.
+    last_child = Category.where(ancestry: parent_id).order('categories.order desc').limit(1)[0]
```
Comment (jonpage) [inline app/models/category.rb]: instead of `.limit(1)[0]`, use `.first`

File: app/models/category.rb (hunk 2 above)
Comment (jonpage) [inline app/models/category.rb]: Just kidding. I was thinking of data_list_measurements. but we really should change this column to something like `list_order` as well.

**Comment by kotodharma** [comment]
All requested changes made and I think ready to merge now.

**Pull request merged**